source_manager.go

```diff
@@ -141,15 +158,121 @@ func NewSourceManager(an ProjectAnalyzer, cachedir string) (*SourceMgr, error) {
 		}
 	}
 
-	return &SourceMgr{
+	sm := &SourceMgr{
 		cachedir: cachedir,
 		lf:       fi,
 		srcs:     make(map[string]source),
 		srcfuts:  make(map[string]*unifiedFuture),
 		an:       an,
 		dxt:      pathDeducerTrie(),
 		rootxt:   newProjectRootTrie(),
-	}, nil
+		qch:      make(chan struct{}),
+	}
+
+	return sm, nil
+}
+
+// SetUpSigHandling sets up typical signal handling for a SourceMgr. It will
+// register a signal handler to be notified on:
+//
+//  - syscall.SIGINT
+//  - syscall.SIGHUP
+//  - syscall.SIGTERM
+//  - syscall.SIGQUIT
+//  - os.Interrupt
+func SetUpSigHandling(sm *SourceMgr) {
+	sigch := make(chan os.Signal)
```

Comment:
You'll certainly want a buffered channel here, of at least 1. Choose a higher value if you think you could receive another signal whilst the signal handler is processing other signals, but probably one is enough by the looks of what you're doing (gracefully handling shutdown).